File: Rust/1603.rs
/**
 * 1603. Design Parking System
 * https://leetcode.com/problems/design-parking-system/
 */

struct ParkingSystem {
  arr: Vec<i32>,
}

impl ParkingSystem {
  fn new(big: i32, medium: i32, small: i32) -> Self {
    ParkingSystem { arr: [big, medium, small].to_vec() }
  }

  fn add_car(&mut self, car_type: i32) -> bool {
    let car_type = (car_type as usize) - 1;

    if self.arr[car_type] == 0 {
      return false;
    }

    self.arr[car_type] -= 1;

    return true;
  }
}

File: Python/1603.py
# 1603. Design Parking System
# https://leetcode.com/problems/design-parking-system/

class ParkingSystem:
  __arr = []

  def __init__(self, big: int, medium: int, small: int):
    self.arr = [big, medium, small]

  def addCar(self, carType: int) -> bool:
    if (self.arr[carType - 1] == 0):
      return False

    self.arr[carType - 1] -= 1

    return True
